src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -132,12 +183,104 @@ public void Initialize(TelemetryConfiguration configuration)
                             portalSender.Send(traceEvent);
                         }
 
+                        // set up heartbeat
+                        if (this.HeartbeatProvider == null)
```

Comment:
how this can be not `null`? There is a flag in the beginning of this method that ensures it's not called twice ever.

Reply:
Good point, removing the check.